src/test/groovy/org/mockito/release/internal/gradle/PublicationsComparatorPluginTest.groovy

```diff
@@ -75,9 +79,57 @@ class PublicationsComparatorPluginTest extends Specification {
         child.evaluate()
 
         then:
-        def task = (PublicationsComparatorTask) child.getTasks()
-                .getByName(PublicationsComparatorPlugin.COMPARE_PUBLICATIONS_TASK);
+        def task = (DownloadPreviousReleaseArtifactsTask) child.getTasks()
+                .getByName(PublicationsComparatorPlugin.DOWNLOAD_PREVIOUS_RELEASE_ARTIFACTS_TASK);
+
+        task.previousVersionPomUrl.contains("bintray.com")
+        task.previousVersionSourcesJarUrl.contains("bintray.com")
+    }
+
+    def "leaves previousVersionPomLocalFile and previousVersionSourcesJarLocalFile null if BintrayPlugin is NOT applied"() {
+        given:
+        def parent = new ProjectBuilder().withProjectDir(tmp.root).withName("parent").build()
+        def child = new ProjectBuilder().withName("child").withParent(parent).build()
+
+        when:
+        child.plugins.apply("org.mockito.mockito-release-tools.publications-comparator")
+        child.evaluate()
+
+        then:
+        def task = (DownloadPreviousReleaseArtifactsTask) child.getTasks()
```

Comment:
You are right that using cast here is technically not necessary (Groovy is dynamic!).

I think @wwilk used it to get better IDEA support, easier to work with the code. Otherwise you don't get IDE suggestions and searches are less smart. 

I'd say it is a good idea to use casting in those situations.